Signal/src/contact/SystemContactsFetcher.swift

```diff
@@ -33,15 +328,27 @@ class SystemContactsFetcher: NSObject {
         return self.authorizationStatus == .authorized
     }
 
-    private let contactStore = CNContactStore()
     private var systemContactsHaveBeenRequestedAtLeastOnce = false
-    private let allowedContactKeys: [CNKeyDescriptor] = [
-        CNContactFormatter.descriptorForRequiredKeys(for: .fullName),
-        CNContactThumbnailImageDataKey as CNKeyDescriptor, // TODO full image instead of thumbnail?
-        CNContactPhoneNumbersKey as CNKeyDescriptor,
-        CNContactEmailAddressesKey as CNKeyDescriptor,
-        CNContactViewController.descriptorForRequiredKeys()
-    ]
+    private var hasSetupObservation = false
+
+    override init() {
+        self.contactStoreAdapter = ContactStoreAdapter()
+    }
+
+    var supportsContactEditing: Bool {
+        return self.contactStoreAdapter.supportsContactEditing
+    }
+
+    private func setupObservationIfNecessary() {
```

Comment:
This is actually an ancillary change. We don't want to start observing on init (because the user may not have authorized access yet)

We want to start observing at "first use" which could be either on launch via `fetchIfAlreadyAuthorized` or a couple different places that potentially call `requestOnce`